cdap-common/src/main/resources/cdap-default.xml

```diff
@@ -678,6 +678,22 @@
     </description>
   </property>
 
+  <property>
+    <name>data.local.storage.user.table.max.open.files</name>
+    <value>100</value>
+    <description>
+      Maximum number of open file descriptors per user table to be used when in CDAP Local Sandbox.
+    </description>
+  </property>
+
+  <property>
+    <name>data.local.storage.system.table.max.open.files</name>
+    <value>500</value>
+    <description>
+      Maximum number of open file descriptors per system table to be used when in CDAP Local Sandbox.
```

Comment:
Hoe many system tables are there? like, 2 dozen? That is 10K+ open files just for system tables?

Reply:
Yea, there are about two dozen. I didn't want to lower this number too much without testing the performance impact of lowering it to a value such as 50 or 100.
Each table won't open 500 files at once; most of the tables' usage remains well below the limit anyways, so we won't hit 10k+ open files. Most likely, just a hand full of the tables will reach the limit of 500.

I spoke to @chtyim, who said that we will need to increase the max open file limit for the CDAP process to a value above the current 1k or 4k that exists on linux system.

Reply:
ok, and as long as it is configurable we can deal with it if it happens